src/v2/providers/testLab.ts

```diff
@@ -0,0 +1,213 @@
+
+  /** URI to the test results in the Firebase Web Console. */
+  resultsUri: string;
+
+  /**
+   * Location in Google Cloud Storage where test results are written to.
+   * In the form "gs://bucket/path/to/somewhere".
+   */
+  gcsPath: string;
+}
+
+/** Information about the client which invoked the test. */
+export interface ClientInfo {
+  /** Client name, such as "gcloud". */
+  client: string;
+
+  /** Map of detailed information about the client. */
+  details: Record<string, string>;
+}
+
+/** The data within all Firebase test matrix completed events. */
+export interface TestMatrixCompletedData {
+  /** Time the test matrix was created. */
+  createTime: string;
+
+  /** State of the test matrix. */
+  state: TestState;
+
+  /**
+   * Code that describes why the test matrix is considered invalid. Only set for
+   * matrices in the INVALID state.
+   */
+  invalidMatrixDetails: string;
+
+  /** Outcome summary of the test matrix. */
+  outcomeSummary: OutcomeSummary;
+
+  /** Locations where test results are stored. */
+  resultStorage: ResultStorage;
+
+  /** Information provided by the client that created the test matrix. */
+  clientInfo: ClientInfo;
+
+  /** ID of the test matrix this event belongs to. */
+  testMatrixId: string;
+}
+
+/**
+ * Event handler which triggers when a Firebase test matrix completes.
+ *
+ * @param handler - Event handler which is run every time a Firebase test matrix completes.
+ * @returns A Cloud Function that you can export and deploy.
+ * @alpha
+ */
+export function onTestMatrixCompleted(
+  handler: (event: CloudEvent<TestMatrixCompletedData>) => any | Promise<any>
+): CloudFunction<CloudEvent<TestMatrixCompletedData>>;
+
+/**
+ * Event handler which triggers when a Firebase test matrix completes.
+ *
+ * @param opts - Options that can be set on an individual event-handling function.
+ * @param handler - Event handler which is run every time a Firebase test matrix completes.
+ * @returns A Cloud Function that you can export and deploy.
+ * @alpha
+ */
+export function onTestMatrixCompleted(
+  opts: EventHandlerOptions,
+  handler: (event: CloudEvent<TestMatrixCompletedData>) => any | Promise<any>
+): CloudFunction<CloudEvent<TestMatrixCompletedData>>;
+
+/**
+ * Event handler which triggers when a Firebase test matrix completes.
+ *
+ * @param optsOrHandler - Options or an event handler.
+ * @param handler - Event handler which is run every time a Firebase test matrix completes.
+ * @returns A Cloud Function that you can export and deploy.
+ * @alpha
+ */
+export function onTestMatrixCompleted(
+  optsOrHandler:
+    | EventHandlerOptions
+    | ((event: CloudEvent<TestMatrixCompletedData>) => any | Promise<any>),
+  handler?: (event: CloudEvent<TestMatrixCompletedData>) => any | Promise<any>
+): CloudFunction<CloudEvent<TestMatrixCompletedData>> {
+  if (typeof optsOrHandler === "function") {
+    handler = optsOrHandler as (event: CloudEvent<TestMatrixCompletedData>) => any | Promise<any>;
```

Comment:
Any chance we can change this to not be any | Promise ? (same with others)

Reply:
I guess my main question was along the lines of if we could avoid the `any`.

But if theres prior art, and would require api review again... I'm alright leave it as-is.